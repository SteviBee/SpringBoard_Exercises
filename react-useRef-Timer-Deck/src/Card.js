import React, { useState } from "react";
// import axios from "axios";

const Card = ({data, handleDraw}) => {
    // creating hid / unhid for button
    const [startButton, setStartButton] = useState(false)
    function toggle(evt) {
        setStartButton(isHidden => !isHidden)
    }

    return (
        // <div>Card are real! {data.map(i => <p>i</p>)}</div>
        <div>Card are real! 
            <img src={data} alt="card1_of_52"></img>
            <button onClick={handleDraw}>Click Me for next card</button>
            <img src={data} alt="card1_of_52"></img>
            <div onClick={toggle}>
                {startButton ? "Cool" : "show timer   "}Start Timer</div>
        </div>
    )

    
}


export default Card;