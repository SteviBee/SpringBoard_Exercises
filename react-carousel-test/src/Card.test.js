import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

// smoke test
it("renders without crashing!", function () {
    render(<Card />)
})

// Snapshot Test
it("matches snapshot", function () {
    // destructure asfragment from render obj of component:
    const { asFragment } = render(<Card caption="test" src="test2" currNum={1} totalNum={4}/>)
    expect(asFragment()).toMatchSnapshot() 
  })

