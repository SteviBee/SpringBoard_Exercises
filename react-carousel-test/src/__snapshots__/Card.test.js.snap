// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test
    </h4>
    <img
      alt="test"
      class="Card-image"
      src="test2"
    />
    <small
      class="Card-small"
    >
      Image 1 of 4.
    </small>
  </div>
</DocumentFragment>
`;
